pkg/resourcemanager/mysql/server/reconcile.go

```diff
@@ -190,9 +190,9 @@ func (m *MySQLServerClient) Ensure(ctx context.Context, obj runtime.Object, opts
 			instance.Status.Provisioned = false
 			instance.Status.FailedProvisioning = true
 			return true, nil
```

Comment:
if this path was taken, wouldn't it mean that we're "done" and no future reconciliations will happen until the spec is changed again?

Just trying to get it straight in my head what exactly was happening here that caused things to get stuck and the only way I see is by taking this path (since all others return false which means keep trying?)

Reply:
but if that's the case, your fix won't fix it because this path would still get taken and then we wouldn't come around again?

Do we need to determine the specific error that's being hit and put it into the known errors collection so that we go around again?

Reply:
No, this error handling wasn't the path that caused the problem. The `CreateServerIfValid` call was returning no error, skipping the `if err != nil` and setting `Provisioning=true, Message="request submitted to Azure"`, returning `false, nil`. Then the next time through it would go down the `if instance.Status.Provisioning` branch, the GetServer on line 84 would return a 404 error but because there's no polling URL nothing would happen - no state change, and as long as the server never appeared the reconciliation would just keep happening forever.

Reply:
Doh, now I get it!